Add blog for deprecation of Media player constants (#2332)
home-assistant/developers.home-assistant: blog/2024-09-23-constants-media-player-deprecation.md

---
author: G Johansson
authorURL: https://github.com/gjohansson-ST
authorImageURL: https://avatars.githubusercontent.com/u/62932417?v=4
authorTwitter: GJohansson

> title: "Start deprecating constants for Media player"

Reviewer: ```suggestion
title: "Deprecating constants for Media player"
```

---

As of Home Assistant Core 2022.5, the feature flag constants used in `MediaPlayerEntity` were deprecated and replaced by the `MediaPlayerEntityFeature` enum. Later, in 2022.10, the repeat mode, media type, and media class constants were deprecated and replaced by `RepeatMode`, `MediaType`, and `MediaClass`, respectively.

> However, no proper deprecation was done, so we start the one-year deprecation period in 2024.10, and the constants will stop working from 2025.10 to ensure all custom integration authors have time to adjust.

Reviewer: ```suggestion
However, no proper deprecation was done, so now in 2024.10 we start the one-year deprecation period, and the constants will stop working from 2025.10, to ensure all custom integration authors have time to adjust.
```

> As of Home Assistant Core 2022.5, the feature flag constants used in `MediaPlayerEntity` was deprecated and replaced by the `MediaPlayerEntityFeature` enum, and later in 2022.10, the repeat mode, media type and media class constants was deprecated and replaced by `RepeatMode`, `MediaType` and `MediaClass` respectively.
> 
> However, there was no proper deprecation done and therefore we start the one-year deprecation period in 2024.10, and the the constants will stop to work from 2025.10 to ensure all custom integration authors have time to adjust.

Reviewer 2: ```suggestion
As of Home Assistant Core 2022.5, the feature flag constants used in `MediaPlayerEntity` were deprecated and replaced by the `MediaPlayerEntityFeature` enum. Later, in 2022.10, the repeat mode, media type, and media class constants were deprecated and replaced by `RepeatMode`, `MediaType`, and `MediaClass`, respectively.

However, no proper deprecation was done, so we start the one-year deprecation period in 2024.10, and the constants will stop working from 2025.10 to ensure all custom integration authors have time to adjust.
```

authorURL: https://github.com/gjohansson-ST
authorImageURL: https://avatars.githubusercontent.com/u/62932417?v=4
authorTwitter: GJohansson

> title: "Deprecating constants for Media player"

Reviewer: ```suggestion
title: "Deprecating constants for Media Player"
```
